packages/lib/shared/utils/error-filters.ts

```diff
@@ -44,3 +44,15 @@ export function isPausedError(error?: Error | null): boolean {
 export function isPausedErrorMessage(errorMessage: string): boolean {
   return errorMessage.includes('reverted with the following reason:\nBAL#402\n')
 }
+
+export function isUnbalancedJoinError(error?: Error | null): boolean {
+  if (!error) return false
+  return isUnbalancedJoinErrorMessage(error)
+}
+
+export function isUnbalancedJoinErrorMessage(error: Error | null): boolean {
+  const errorStrings = ['BAL#304', 'queryAddLiquidityUnbalanced'] // [v2 error, v3 error]
+  const hasErrors = (errorString: string) => error?.message.includes(errorString)
+
+  return errorStrings.some(hasErrors)
+}
```

Comment:
I know it's more verbose, but we're supposed to be using Add or AddLiquidity instead Join terminology.

Reply:
Renamed!